pkg/network/deviceinfo/deviceinfo.go

```diff
@@ -33,11 +33,11 @@ func MapNetworkNameToDeviceInfo(networks []v1.Network,
 	networkStatuses []networkv1.NetworkStatus,
 ) map[string]*networkv1.DeviceInfo {
 	multusInterfaceNameToNetworkStatus := multus.NetworkStatusesByPodIfaceName(networkStatuses)
```

Comment:
can't we get rid of this line?
we can replace `networkStatusEntry, exist := multusInterfaceNameToNetworkStatus[multusInterfaceName]` with a [binary search](https://pkg.go.dev/slices#BinarySearchFunc) over the `interfaces` slice

Reply:
From the binarySearch docs:
> The slice must be sorted in increasing order, where "increasing" is defined by cmp

To the best of my knowledge, the `networkv1.NetworkStatus` slices are not ordered in a way that a binary search can utilize.
I encourage you to explore this subject in a follow-up.

Reply:
right :( then we can use https://pkg.go.dev/slices#IndexFunc, this slice has got to be very small, so it doesn't really matter how we search through it

Reply:
@nirdothan , I think it is how one defines elegant code.
If we go over a list and we search again and again through it, it becomes a N^2 cost.

While N is small, this does not really matter in terms of performance, but it does cause it to look "ugly".

So for the sake of an elegant code, we pass the whole list, index per what we need and then go over some loop and check existence. For most, this is seen as more elegant, but it may be just a subjective thing that we got used to. Personally I find searching again and again in the same list ugly, no matter the size of the list.

Reply:
> I'm not so happy about making another traversal

It is one traversal vs N traversals, so I do not know why you see it as "making another".
That is exactly why I think it is ugly and not elegant.

> and building yet another map just to later check if an element exists in a slice.

Yes, that is the cost, to convert one data structure to another in order to efficiently lookup entries.
That new data structure has a memory cost, but as you pointed out, is negligible due to the size.

> in HL programming languages such as Python or Java you would never do that.

That is because those languages usually have dedicated data structures which are optimized for searching in the first place.

> I could research multiple solutions that are more elegant, maybe by changing the type/s returned from `CreateFromNetworkStatuses()`.

At its base, that means you either can avoid the search, use a better data structure than slice, etc.